scalableminds/webknossos
# Slider improvements

### URL of deployed dev instance (used for testing):
- https://sliderimprovements.webknossos.xyz

### Steps to test:
- Almost every slider in webknossos should react to wheel actions, meaning that wheeling while hovering above it should change the value, if the slider is not disabled. Also, double-clicking the knob (or one of the knobs for range sliders) should reset it to its default value.
- If possible, test with mouse and touchpad to ensure both interactions feel alright (e.g. step/"speed" of the slider)
- This is true for range sliders and sliders with a single value
- A kind of slider for which it doesnt work are MagSliders
- A good place to start testing is the layer settings tab on the left

### TODOs:
- [x] test all sliders
     - (done except node radius)
- [x] pass through step size
- [x] for range: make range smaller or bigger
- [x] register click on knob?
- [x] make default values/initial values accessible for sliders
### Special candidates
- [x] MagSlider in download_modal_view doesnt wheel
- [x] Slider in quick_select_settings dont wheel well 
- [x] flight options: Mouse Rotation, Keyboard Rotation, Crosshair Size
- [x] layer settings: node radius not editable, how to do that?; particle size onWheel

### Issues:
- fixes #7266

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Updated [migration guide](../blob/master/MIGRATIONSwhom.unreleased.md#unreleased) if applicable
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md) dont think these are applicable here because onChange method is not altered, and only click and wheel handling are altered
